AdvancedDLSupport/ImplementationGenerators/Wrappers/SpanMarshallingWrapper.cs

```diff
@@ -62,21 +64,52 @@ ImplementationOptions options
         /// <inheritdoc />
         public override IntrospectiveMethodInfo GeneratePassthroughDefinition(PipelineWorkUnit<IntrospectiveMethodInfo> workUnit)
         {
-            if (!workUnit.Definition.ReturnType.GenericTypeArguments[0].IsValueType)
+            Type returnType = workUnit.Definition.ReturnType, newReturnType;
+            var definition = workUnit.Definition;
+
+            if (IsSpanType(returnType))
             {
-                // Span<byte> is used because unbound generics are not allowed inside a nameof, and it still results as just 'Span'
-                throw new NotSupportedException($"Method is not a {nameof(ValueType)} and cannot be marshaled as a {nameof(Span<byte>)}. Marshalling {nameof(Span<byte>)}" +
-                                                $"requires the marshaled type T in {nameof(Span<byte>)}<T> to be a {nameof(ValueType)}");
-            }
+                var genericType = returnType.GenericTypeArguments[0];
 
-            var definition = workUnit.Definition;
+                if (IsOrContainsReferences(genericType))
+                {
+                    // Span<byte> is used because unbound generics are not allowed inside a nameof, and it still results as just 'Span'
+                    throw new NotSupportedException($"Method Return Type is a class or contains references to classes and cannot be marshaled as a {nameof(Span<byte>)}. Marshalling {nameof(Span<byte>)}" +

```

Comment:
Instead of using nameof, you should probably just pull the name from the return type itself.